R/method-differentialAbundance.R

```diff
@@ -0,0 +1,170 @@
+      #             fix_formula = comparisonVariable, rand_formula = NULL,
+      #             p_adj_method = "holm", pseudo = 0, pseudo_sens = TRUE,
+      #             prv_cut = 0.10, lib_cut = 0, s0_perc = 0.05,
+      #             group = comparisonVariable, struc_zero = FALSE, neg_lb = FALSE,
+      #             alpha = 0.05, n_cl = 2, verbose = TRUE,
+      #             global = FALSE, pairwise = FALSE, 
+      #             dunnet = FALSE, trend = FALSE,
+      #             iter_control = list(tol = 1e-5, max_iter = 20, 
+      #                                 verbose = FALSE),
+      #             em_control = list(tol = 1e-5, max_iter = 100),
+      #             lme_control = NULL, mdfdr_control = NULL, 
+      #             trend_control = NULL)
+
+    print('good try!!')
+
+    } else {
+      stop('Unaccepted dissimilarity method. Accepted methods are bray, jaccard, and jsd.')
+    }
+    
+    ## Make the result a Compute result. Return the samples that are left as the data. Return
+    # the rest as statistics
+    result <- new("ComputeResult")
+    result@name <- 'differentialAbundance'
+    result@recordIdColumn <- recordIdColumn
+    result@ancestorIdColumns <- ancestorIdColumns
+    result@statistics <- statistics
+
+    # # Handle errors or return positive computeMessage
+    # if (veupathUtils::is.error(dist)) {
+    #   veupathUtils::logWithTime(paste('Differential abundance computation FAILED with parameters method=', method), verbose)
+    #   stop() 
+    # } else {
+    #   veupathUtils::logWithTime("Computed dissimilarity matrix.", verbose)
+    #   computeMessage <- paste(method, "dissimilarity matrix computation successful.")
+    # }
+
+    entity <- veupathUtils::strSplit(recordIdColumn,".", 4, 1)
+    # result@computationDetails <- paste(computeMessage, ', pcoaVariance =', percentVar)
+    # result@parameters <- paste('method =', method)   
+    
+    
+    # makeVariableMetadataObject <- function(varName) {
+    #   #bit hacky, see if you can think of something better
+    #   plotRef <- ifelse(identical(varName, 'log2FoldChange'), 'xAxis', 'tooltip')
+
+    #   veupathUtils::VariableMetadata(
+    #              variableClass = veupathUtils::VariableClass(value = "computed"),
+    #              variableSpec = veupathUtils::VariableSpec(variableId = varName, entityId = entity),
+    #              plotReference = veupathUtils::PlotReference(value = plotRef),
+    #             #  displayName = displayName, # Ann add logic for display name
+    #              displayRangeMin = min(dt[[axisName]]),
+    #              displayRangeMax = max(dt[[axisName]]),
+    #              dataType = veupathUtils::DataType(value = "NUMBER"),
+    #              dataShape = veupathUtils::DataShape(value = "CONTINUOUS")
+    #   )
+    # }
+
+    ### This doesnot work yet...  
+    # computedStatisticMetadata <- veupathUtils::VariableMetadataList(lapply(displayNames, makeVariableMetadataObject))
```

Comment:
definitely dont need plot references but display names would be helpful for tooltips and tables. Not essential though for the first implementation since the volcano could set lots of that stuff itself